test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarQuery/DollarQueryTests.cs

```diff
@@ -0,0 +1,205 @@
+                        new Tuple<string, string>(CustomersResourcePath, "$orderby=Id desc"),
+                        new Tuple<string, string>(CustomersResourcePath, "$top=1"),
+                        new Tuple<string, string>(CustomersResourcePath, "$top=1&$skip=3"),
+                        new Tuple<string, string>(CustomersResourcePath, "$orderby=Id desc&top=2&skip=3"),
+                        new Tuple<string, string>(CustomersResourcePath, "$select=Id,Name"),
+                        new Tuple<string, string>(CustomersResourcePath, "$expand=Orders"),
+                        new Tuple<string, string>(CustomersResourcePath, "$select=Orders&$expand=Orders"),
+                        new Tuple<string, string>(CustomersResourcePath, "$format=" + Uri.EscapeDataString(ApplicationJsonODataMinimalMetadataStreamingFalse)),
+                        new Tuple<string, string>(CustomersResourcePath, "$expand=SpecialOrder($select=Detail)&$filter=Id le 5&$orderby=Id desc&$select=Id&$format=" + Uri.EscapeDataString(ApplicationJsonODataMinimalMetadataStreamingFalse)),
+                        new Tuple<string, string>(SingleCustomerResourcePath, "$select=Id,Name"),
+                        new Tuple<string, string>(SingleCustomerResourcePath, "$expand=Orders"),
+                        new Tuple<string, string>(SingleCustomerResourcePath, "$format=" + Uri.EscapeDataString(ApplicationJsonODataMinimalMetadataStreamingFalse)),
+                        new Tuple<string, string>(SingleCustomerResourcePath, "$expand=SpecialOrder($select=Detail)&$select=Id&$format=" + Uri.EscapeDataString(ApplicationJsonODataMinimalMetadataStreamingFalse))
+                    })
+                {
+                    odataQueryOptionsData.Add(tuple.Item1, tuple.Item2);
+                }
+
+                return odataQueryOptionsData;
+            }
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            configuration.Routes.Clear();
+            configuration.Count().Filter().OrderBy().Expand().MaxTop(null).Select();
+            configuration.MapODataServiceRoute("odata", "odata", GetEdmModel(configuration), new DefaultODataPathHandler(), ODataRoutingConventions.CreateDefault());
+        }
+
+        private static IEdmModel GetEdmModel(WebRouteConfiguration configuration)
+        {
+            ODataConventionModelBuilder builder = configuration.CreateConventionModelBuilder();
+
+            EntitySetConfiguration<DollarQueryCustomer> dollarQueryCustomers =
+                builder.EntitySet<DollarQueryCustomer>("DollarQueryCustomers");
+
+            EntitySetConfiguration<DollarQueryOrder> dollarQueryOrders =
+                builder.EntitySet<DollarQueryOrder>("DollarQueryOrders");
+
+            return builder.GetEdmModel();
+        }
+
+        [Theory]
+        [MemberData(nameof(ODataQueryOptionsData))]
```

Comment:
>[MemberData(nameof(ODataQueryOptionsData))] [](start = 8, length = 43)

add a variant where there are some query options specified in the body and some specified in the request URL, both where they don't overlap (in which case, they should be merged) and where they do overlap (in which case, an error should be thrown)

Reply:
@mikepizzo Add tests to cover those two scenarios:
https://github.com/OData/WebApi/pull/2191/files#diff-9e480fb8b0555db1128de67981ff5e0aR123
https://github.com/OData/WebApi/pull/2191/files#diff-9e480fb8b0555db1128de67981ff5e0aR142